main.go

```diff
@@ -49,55 +49,66 @@ func run(ctx context.Context) error {
 	if err != nil {
-		log.Event(ctx, "encountered error parsing kafka max bytes", log.FATAL, log.Error(err))
+		log.Fatal(ctx, "encountered error parsing kafka max bytes", err)
 		return err
 	}
 
 	// External services and their initialization state
 	var serviceList initialise.ExternalServiceList
 
-	syncConsumerGroup, err := serviceList.GetConsumer(ctx, cfg)
+	syncConsumerGroup, err := serviceList.GetConsumer(ctx, cfg.KafkaConfig)
 	if err != nil {
-		log.Event(ctx, "could not initialise kafka consumer", log.FATAL, log.Error(err), log.Data{"group": cfg.ConsumerGroup, "topic": cfg.ConsumerTopic})
+		log.Fatal(ctx, "could not initialise kafka consumer", err, log.Data{"group": cfg.KafkaConfig.ConsumerGroup, "topic": cfg.KafkaConfig.ConsumerTopic})
 		return err
 	}
 
-	searchBuiltProducer, err := serviceList.GetProducer(ctx, cfg.Brokers, cfg.ProducerTopic, initialise.SearchBuilt, int(envMax), cfg)
+	searchBuiltProducer, err := serviceList.GetProducer(ctx, cfg.KafkaConfig, cfg.KafkaConfig.ProducerTopic, initialise.SearchBuilt, int(envMax))
 	if err != nil {
-		log.Event(ctx, "could not initialise kafka producer", log.FATAL, log.Error(err), log.Data{"topic": cfg.ProducerTopic})
+		log.Fatal(ctx, "could not initialise kafka producer", err, log.Data{"topic": cfg.KafkaConfig.ProducerTopic})
 		return err
 	}
 
-	searchBuilderErrProducer, err := serviceList.GetProducer(ctx, cfg.Brokers, cfg.EventReporterTopic, initialise.SearchBuilderErr, int(envMax), cfg)
+	searchBuilderErrProducer, err := serviceList.GetProducer(ctx, cfg.KafkaConfig, cfg.KafkaConfig.EventReporterTopic, initialise.SearchBuilderErr, int(envMax))
 	if err != nil {
-		log.Event(ctx, "could not initialise kafka producer", log.FATAL, log.Error(err), log.Data{"topic": cfg.EventReporterTopic})
+		log.Fatal(ctx, "could not initialise kafka producer", err, log.Data{"topic": cfg.KafkaConfig.EventReporterTopic})
 		return err
 	}
 
 	// Get Error reporter
 	errorReporter, err := serviceList.GetImportErrorReporter(searchBuilderErrProducer, log.Namespace)
 	if err != nil {
-		log.Event(ctx, "error while attempting to create error reporter client", log.FATAL, log.Error(err))
+		log.Fatal(ctx, "error while attempting to create error reporter client", err)
 		return err
 	}
 
 	// Get HealthCheck
 	hc, err := serviceList.GetHealthCheck(cfg, BuildTime, GitCommit, Version)
 	if err != nil {
-		log.Event(ctx, "could not instantiate healthcheck", log.FATAL, log.Error(err))
+		log.Fatal(ctx, "could not instantiate healthcheck", err)
 		return err
 	}
 
 	hierarchyClient := hierarchy.New(cfg.HierarchyAPIURL)
-	elasticSearchClient := elasticsearch.NewClient(cfg.ElasticSearchAPIURL, cfg.SignElasticsearchRequests, cfg.MaxRetries)
+
+	var awsSDKSigner *esauth.Signer
+	if cfg.SignElasticsearchRequests {
+		awsSDKSigner, err = esauth.NewAwsSigner("", "", cfg.AwsRegion, cfg.AwsService)
+		if err != nil {
+			log.Error(ctx, "failed to create aws v4 signer", err)
+			os.Exit(1)
```

Comment:
This should return err
```suggestion
			return err
```

Reply:
https://github.com/ONSdigital/dp-dimension-search-builder/pull/52

Reply:
this is now addressed